framework/vst/vstscaner.cpp

```diff
@@ -0,0 +1,63 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "vstscaner.h"
+
+#include <QDir>
+#include <QString>
+#include "pluginloader.h"
+
+using namespace mu::vst;
+
+VSTScaner::VSTScaner(std::string paths)
+    : m_paths(paths), m_plugins()
+{
+}
+
+void VSTScaner::scan()
+{
+    QString dirPaths = QString::fromStdString(m_paths);
+    for (auto dirPath : dirPaths.split(";")) {
+        QDir directory(dirPath);
+        if (directory.exists()) {
+            for (auto pluginName : directory.entryList(QStringList({"*.vst3"}))) {
```

Comment:
We have fsoperations service, it is better to use it to work with the file system.

Reply:
fsoperations have not functionality for scanning modules (plugin.vst3 is not a file)